Framework/Geometry/src/Instrument/InstrumentVisitor.cpp

```diff
@@ -309,5 +310,18 @@ InstrumentVisitor::makeWrappers() const {
   return {std::move(compInfoWrapper), std::move(detInfoWrapper)};
 }
 
+std::pair<std::unique_ptr<ComponentInfo>, std::unique_ptr<DetectorInfo>>
+InstrumentVisitor::makeWrappers(const Instrument &instrument,
+                                ParameterMap *pmap) {
+  const auto parInstrument =
+      pmap ? ParComponentFactory::createInstrument(
+                 boost::shared_ptr<const Instrument>(&instrument, NoDeleting()),
+                 boost::shared_ptr<ParameterMap>(pmap, NoDeleting()))
```

Comment:
I need a `shared_ptr` to create `parInstrument`, but `InstrumentVisitor::makeWrappers` is not the owner of `instrument` and `pmap`, so we have to use an allocator that does not delete them.